apsuite/loco/config.py

```diff
@@ -431,11 +432,28 @@ def update_sext_knobs(self):
                 self.sext_indices = []
                 for fam_name in self.sextupoles_to_fit:
                     self.sext_indices += self.respm.fam_data[fam_name]['index']
-                    self.sext_indices.sort()
+                self.sext_indices.sort()
 
     def update_skew_quad_knobs(self):
         """."""
-        self.skew_quad_indices = self.respm.fam_data['QS']['index']
+        """."""
+        if self.skew_quadrupoles_to_fit is None:
+            self.skew_quad_indices = self.respm.fam_data['QS']['index']
+        else:
+            skewquadfit = set(self.skew_quadrupoles_to_fit)
+            skewquadall = set(self.famname_skewquadset)
+            if not skewquadfit.issubset(skewquadall):
+                raise Exception('invalid skew quadrupole name used to fit!')
+            else:
+                self.skew_quad_indices = []
+                for fam_name in self.skew_quadrupoles_to_fit:
+                    self.skew_quad_indices += self.respm.fam_data[
+                        fam_name]['index']
+                idx_all = _np.array(
+                    self.respm.fam_data['QS']['index']).flatten()
+                idx_sub = _np.array(self.skew_quad_indices).flatten()
+                self.skew_quad_indices = list(set(idx_sub) & set(idx_all))
```

Comment:
Yes I am assuming no segmentation for QS. I am considering that the only segmented elements in the lattice are the dipoles and dealing with them accordingly. I concluded that since the re-factorization that we discussed is planned to happen ASAP and it will generalize the code and make it more robust against possible problems of that kind, if the question is not a bug, then it is not worth to spend much time on it. I truly expect that the re-factorization will happen sooner than significant modifications in the lattice, like changing the elements segmentation.

Reply:
ok!